libraries/chain/controller.cpp

```diff
@@ -2688,11 +2690,30 @@ struct controller_impl {
       else
 #endif
          // Non-EOSVMOC needs a wasmif per thread
-         wasmif_thread_local = std::make_unique<wasm_interface>( conf.wasm_runtime, conf.eosvmoc_tierup, db, conf.state_dir, conf.eosvmoc_config, !conf.profile_accounts.empty());
+         threaded_wasmifs[std::this_thread::get_id()] = std::make_unique<wasm_interface>( conf.wasm_runtime, conf.eosvmoc_tierup, db, conf.state_dir, conf.eosvmoc_config, !conf.profile_accounts.empty());
```

Comment:
This is called from multiple threads at startup and it is not thread safe. Since it is only modified during start of the threads you can likely just use a mtx only here. 

Also to be extra cautious you should probably wait in `producer_plugin::startup()` until all `init_thread_local_data()` have been called. You could use some futures in `producer_plugin::startup()` to do that. That seems extremely unlikely, but technically possible I think.

Reply:
Thanks! I forgot about the mutex here.